pkg/core/blockchain.go

```diff
@@ -1209,9 +1171,20 @@ func (bc *Blockchain) verifyTx(t *transaction.Transaction, block *block.Block) e
 	if t.ValidUntilBlock <= height || t.ValidUntilBlock > height+transaction.MaxValidUntilBlockIncrement {
 		return errors.Errorf("transaction has expired. ValidUntilBlock = %d, current height = %d", t.ValidUntilBlock, height)
 	}
-	if io.GetVarSize(t) > transaction.MaxTransactionSize {
+	balance := util.Fixed8FromInt64(bc.GetNEP5Balances(t.Sender).Trackers[bc.contracts.GAS.Hash].Balance)
+	need := t.SystemFee.Add(t.NetworkFee)
+	if balance.LessThan(need) {
+		return errors.Errorf("insufficient funds: balance is %v, need: %v", balance, need)
+	}
+	size := io.GetVarSize(t)
+	if size > transaction.MaxTransactionSize {
 		return errors.Errorf("invalid transaction size = %d. It shoud be less then MaxTransactionSize = %d", io.GetVarSize(t), transaction.MaxTransactionSize)
 	}
+	needNetworkFee := int64(size) * bc.FeePerByte().IntegralValue()
```

Comment:
Why `IntegralValue()`?! At the moment that would be 0.

Reply:
So, which one should I use here, if not `IntegralValue` and not `FloatValue`?

Reply:
`int64(bc.FeePerByte())*int64(size)`?

Reply:
Yes, it suits

Reply:
Fixed this.